core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala

```diff
@@ -400,9 +400,15 @@ trait WorkflowComponent {
       }
 
     def listWorkflowRecsForSubmissionAndStatuses(submissionId: UUID,
+                                                 limit: Int,
                                                  statuses: WorkflowStatuses.WorkflowStatus*
     ): ReadAction[Seq[WorkflowRecord]] =
-      findWorkflowsBySubmissionId(submissionId).filter(_.status inSetBind (statuses.map(_.toString))).result
+      findWorkflowsBySubmissionId(submissionId).filter(_.status inSetBind (statuses.map(_.toString))).take(limit).result
+
+    def countWorkflowRecsForSubmissionAndStatuses(submissionId: UUID,
+                                                  statuses: WorkflowStatuses.WorkflowStatus*
+    ): ReadAction[Int] =
+      findWorkflowsBySubmissionId(submissionId).filter(_.status inSetBind (statuses.map(_.toString))).length.result
```

Comment:
previously, `SubmissionMonitorActor.updateSubmissionStatus()` called `listWorkflowRecsForSubmissionAndStatuses()`, returning all workflow records as a Seq, and then checked if that Seq was empty or not. This moves the logic to the db, returning just a count. I needed to change *something* here, because by adding a limit to `listWorkflowRecsForSubmissionAndStatuses()`, it would have been invalid for `SubmissionMonitorActor.updateSubmissionStatus()` - two birds with one stone?